src/json-validator.cpp

```diff
@@ -727,7 +727,7 @@ class string : public schema
 #ifndef NO_STD_REGEX
 		attr = sch.find("pattern");
 		if (attr != sch.end()) {
-			patternString_ = attr.value();
+			patternString_ = attr.value().get<std::string>();
```

Comment:
What is the error message you're receiving for this `get<>()`-call? Which version of the compiler are you using?

I'm asking because the compiler should determine automatically the requested type.

Reply:
I tried to recreate the issue, but it seems that using v3.8.0 nlohmann json.hpp and latest commit it works.
I'll remove this and check into adding MSVC & Clang compilers into CI to minimize this kind of errors in future.

For reference: 
Using v3.7.3 json.hpp and commit (a0fca479f60235ed775b7aac8c441688466c3a76) on Visual Studio 2019 it produces the following error: 
```
2> ...\json-schema-validator\src\json-validator.cpp(660,1): error C2593: 'operator =' is ambiguous
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\xstring(2591,19): message : could be 'std::basic_string<char,std::char_traits<char>,std::allocator<char>> &std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator =(std::initializer_list<_Elem>)'
2>        with
2>        [
2>            _Elem=char
2>        ]
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\xstring(2492,19): message : or       'std::basic_string<char,std::char_traits<char>,std::allocator<char>> &std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator =(std::basic_string<char,std::char_traits<char>,std::allocator<char>> &&) noexcept(<expr>)'
2> ...\json-schema-validator\src\json-validator.cpp(660,1): message : while trying to match the argument list '(std::string, nlohmann::basic_json<std::map,std::vector,std::string,bool,int64_t,uint64_t,double,std::allocator,nlohmann::adl_serializer>)'
2>json-patch.cpp
```